### DISCUSSION TITLE
How to stop feed run after a time window

### DISCUSSION THREAD

**Discussion by sbkhosh** [bmoscon/cryptofeed#895] [Q&A] [answered]
Hello,
I am still learning the features of this package. How is it possible to stop the feed after say 30 minutes, in order to collect the data stream only during this period of time. I am doing something like this, which writes for all my pairs to csv, but would like to stop the stream after 30 minutes (besides the obvious possibility of Ctrl+C in terminal).

    async def ohlcv(self,data):
        if len(data) != 0:
            for el in self.cryptfd_pairs:
                data[el]['timestamp'] = int(time.time())
                data[el]['vwap'] = round(data[el]['vwap'],3)
            res = pd.DataFrame(data).T
            res.reset_index(inplace=True)
            res.rename({'index': 'coin'},inplace=True)
            res.to_csv(self.cryptfd['filename_store'],header=False,mode='a')

    def get_feed_handler(self):
        init_time = time.time()
        f = FeedHandler()
        f.add_feed(BinanceFutures(symbols=self.cryptfd_pairs,
                                  channels=[TRADES],
                                  callbacks={TRADES: OHLCV(self.ohlcv, window=self.cryptfd['time_window'])}))
        f.run()

Thank you

**Comment by bmoscon** [comment, marked as answer]
yes, you can do this:

https://github.com/bmoscon/cryptofeed/blob/master/tools/generate_test_data.py#L66

**Reply by sbkhosh** [reply]
Thanks very much
